rust_src/src/lread.rs

```diff
@@ -404,4 +405,52 @@ pub fn read_char_exclusive(
     }
 }
 
+/// Delete the symbol named NAME, if any, from OBARRAY.
+/// The value is t if a symbol was found and deleted, nil otherwise.
+/// NAME may be a string or a symbol.  If it is a symbol, that symbol
+/// is deleted, if it belongs to OBARRAY--no other symbol is deleted.
+/// OBARRAY, if nil, defaults to the value of the variable `obarray'.
+/// usage: (unintern NAME OBARRAY)
+#[lisp_fn(min = "1")]
+pub fn unintern(name: LispSymbolOrString, obarray: Option<LispObarrayRef>) -> LispObject {
+    let obarray = obarray.unwrap_or_else(LispObarrayRef::global);
+    let obarray: LispObarrayRef = check_obarray(obarray.into()).into();
+
+    let tem = obarray.lookup(name);
+    if tem.is_integer() {
+        return Qnil;
+    }
+    // If arg was a symbol, don't delete anything but that symbol itself.
+    if name.is_symbol() && name != tem {
+        return Qnil;
+    }
+
+    let mut temp: LispSymbolRef = tem.into();
+    temp.set_uninterned();
+
+    let hash = unsafe { oblookup_last_bucket_number };
+
```

Comment:
If we add `get` and `set` methods to `LispObarrayRef` this code becomes simpler to read and reason about.

Reply:
Sure thing, I can add wrapper methods if that's what you mean. Actually, now I'm wondering. Since every (valid) obarray is also a vector-like... would it make sense to re-use the macro `impl_vectorlike_ref!` for `LispObarrayRef`?

Reply:
If I recall, `LispObarrayRef` is a simple type alias for a `LispObject`. Perhaps it makes more sense for it to be an alias for `LispVector` which would then make adding pass through methods easy. That sounds beyond the scope of this PR but definitely a worthwhile follow on effort.